src/D2L.CodeStyle.Analyzers/Immutability/ImmutableAttributeConsistencyChecker.cs

```diff
@@ -45,7 +46,38 @@ INamedTypeSymbol typeSymbol
 			}
 		}
 
-		public void CompareConsistencyToBaseType(
+		public void CheckMethodDeclaration(
+			IMethodSymbol methodSymbol
+		) {
+			if( methodSymbol.TypeParameters.Length == 0 ) {
+				return;
+			}
+
+			ImmutableArray<IMethodSymbol> implementedMethods = methodSymbol.GetImplementedMethods();
+			foreach( IMethodSymbol implementedMethod in implementedMethods ) {
+				Debug.Assert( methodSymbol.TypeParameters.Length == implementedMethod.TypeParameters.Length );
+

```

Comment:
```suggestion
```

https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debug.assert?view=net-5.0

This pops up a message box on failure. This would be very bad on a build server.
If you want the check then use an exception :)

Reply:
Like, a win32 UI dialog :)

Reply:
Yeah - I thought it did something more reasonable depending on the situation.

Reply:
It's a shame.